File: docs/DSA/neetcode150/src/go/product_of_array_except_self.go
package main

func productExceptSelf(nums []int) []int {
	result := make([]int, len(nums))
	var pre, post int = 1, 1
	for idx := range nums {
		result[idx] = pre
		pre *= nums[idx]
	}

	for idx := len(nums) - 1; idx >= 0; idx-- {
		result[idx] *= post
		post *= nums[idx]
	}
	return result
}

File: docs/DSA/neetcode150/src/py/product_of_array_except_self.py
from typing import List
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = []
        pre, post = 1, 1

        for i in range(len(nums)):
            result.append(pre)
            pre *= nums[i]

        for i in reversed(range(len(nums))):
            result[i] *= post
            post *= nums[i]
        print(result)
        return result
